modules/persistent-storage-csi-snapshots-provision.adoc

```diff
@@ -0,0 +1,16 @@
+// Module included in the following assemblies:
+//
+// * storage/persistent_storage/persistent-storage-csi-snapshots.adoc
+
+[id="persistent-storage-snapshots-provision_{context}"]
+= Provisioning a volume snapshot
+
+There are two ways to provision snapshots: manually and dynamically.
+
+== Manual provisioning
+
+As a cluster administrator, you can manually pre-provision a number of VolumeSnapshotContents. These carry the real volume snapshot details available to cluster users.
+
+== Dynamic provisioning
+
+Instead of using a preexisting snapshot, you can request that a snapshot be taken dynamically from a PersistentVolumeClaim. Attributes are specified using VolumeSnapshotClass.
```

Comment:
In storage class, "parameters" is one of the fields. I see "attribute" used to describe "StorageClass attribute" and "storageClassName attribute" in our docs (e.g., https://docs.openshift.com/container-platform/4.3/storage/understanding-persistent-storage.html#pvc-storage-class_understanding-persistent-storage) and elsewhere. Maybe "attribute" is used because "parameter" is more specific in StorageClass.

Reply:
@bobfuru, looks like a consistency issue; I don't think we ever defined how to write about a YAML file, although generally a key is referred to as a `parameter`. In that example, the attribute is also a `parameter` in the YAML file.

I prefer JSON, because it really is a text representation of an object, and terms like `key` and `value` and `array` are valid and understood. With YAML, I'm not really sure. It's more like a Windows `.ini` file or whatever.

I've wanted to formalize some kind of vocabulary for these, but in the meanwhile I keep defaulting to `parameter` for any key in a YAML file.